### PR TITLE
feat(init): Implement support for Wolfi

### PR DESCRIPTION
**Pull request by EyeCantCU**
Install the Wolfi base package and libc utilities

### CODE REVIEW

**Comment by EyeCantCU** [comment]
Depends on: https://github.com/89luca89/distrobox/pull/1191

**Comment by 89luca89** [comment]
Can we rework this so that it doesn't depend on #1191 ?

From what I understand, we just need that if/else on the base package right?

File: distrobox-init [hunk 1]
```diff
@@ -415,64 +415,111 @@ if [ "${upgrade}" -ne 0 ] ||
...
-			wget
-			which
-			xauth
-			xz
-			zip
-			$(apk search -q mesa-dri)
-			$(apk search -q mesa-vulkan)
-			vulkan-loader
-		"
+		if apk add wolfi-base; then
```
Comment (89luca89) [inline distrobox-init:418]: I would declare the common packages in `deps` outside of the "if" and then just add the packages of difference augmenting the common ones like:

```sh
deps="$deps
foo
bar
baz
"
```

**Review by 89luca89 — CHANGES_REQUESTED** [review]

File: distrobox-init (hunk 1 above)
Comment (EyeCantCU) [reply distrobox-init:418]: Cleaned up the deps lists :)

File: distrobox-init [hunk 2]
```diff
@@ -415,63 +415,74 @@ if [ "${upgrade}" -ne 0 ] ||
...
 		fi
-		deps="
+		if apk add wolfi-base; then
+			deps="
+				busybox
+				gnutar
+				man-db
+				mesa
+				posix-libc-utils
+				sudo-rs
```
Comment (89luca89) [inline distrobox-init:425]: This package conflicts with `util-linux-login` that is needed for a full `su` package

Either we remove sudo-rs or we solve this upstream 

image

**Review by 89luca89 — CHANGES_REQUESTED** [review]

File: distrobox-init [hunk 3]
```diff
@@ -415,63 +415,74 @@ if [ "${upgrade}" -ne 0 ] ||
...
+				pinentry
+				sudo
+				tar
+				vte3
+				which
+				$(apk search -q mesa-dri)
+				$(apk search -q mesa-vulkan)
+			"
+		fi
+		deps="${deps}
```
Comment (89luca89) [inline distrobox-init:446]: it woud be good to change `${deps}` to `${deps:-}` so that we have a default value in case the if/else before fails

**Review by 89luca89 — CHANGES_REQUESTED** [review]

**Review by 89luca89 — APPROVED** [review]

**Comment by 89luca89** [comment]
Great stuff @EyeCantCU  thanks! :heart:

**Comment by EyeCantCU** [comment]
No problem Luca! Happy to help out

**Pull request merged**